Modules/System/Azure Storage Services Authorization/src/ReadySAS/StorServAuthReadySAS.Codeunit.al

```diff
@@ -0,0 +1,52 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 9065 "Stor. Serv. Auth. Ready SAS" implements "Storage Service Authorization"
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    [NonDebuggable]
+    procedure Authorize(var HttpRequestMessage: HttpRequestMessage; StorageAccount: Text)
+    var
+        Uri: Codeunit Uri;
+        UriBuilder: Codeunit "Uri Builder";
+        UriText, QueryText : Text;
+    begin
+        UriText := HttpRequestMessage.GetRequestUri();
+
+        UriBuilder.Init(UriText);
+        QueryText := UriBuilder.GetQuery();
+
+        QueryText := DelChr(QueryText, '<', '?'); // remove ? from the query
+
+        if QueryText <> '' then
+            QueryText += '&';
+        QueryText += GetSharedAccessSignature();
+        UriBuilder.SetQuery(QueryText);
+
+        UriBuilder.GetUri(Uri);
+
+        HttpRequestMessage.SetRequestUri(Uri.GetAbsoluteUri());
+    end;
+
+    [NonDebuggable]
+    procedure GetSharedAccessSignature(): Text
+    begin
+        exit(SharedAccessSignature);
+    end;
+
+    [NonDebuggable]
+    procedure SetSharedAccessSignature(NewSharedAccessSignature: Text)
+    begin
+        SharedAccessSignature := NewSharedAccessSignature;
+        if SharedAccessSignature.StartsWith('?') then
```

Comment:
Does this happen? I'd assume the signature given wouldn't have `?`

Reply:
I wanted to make sure it's always in one standard format, because when you generate the SASToken from azure it does have `?` at the start, like here:
![image](https://github.com/microsoft/ALAppExtensions/assets/40309728/75256a42-072a-4264-93f7-a868f851d500)